AdminDashboard/Controllers/AccountController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using AdminDashboard.BusinessLogicOrchestrators.AccountOrchestrator;
+using AdminDashboard.Models.JsonRequests;
+using AdminDashboard.Validators.Plans;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace AdminDashboard.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class AccountController : ControllerBase
+    {
+        private readonly IAccountOrchestrator Orchestrator;
+        private readonly IPlanValidator PlanValidator;
+
+        public AccountController(IAccountOrchestrator orchestrator,
+            IPlanValidator planValidator)
+        {
+            Orchestrator = orchestrator;
+            PlanValidator = planValidator;
+        }
+
+        /// <summary>
+        /// Gets an account by its id
+        /// </summary>
+        /// <param name="id">The unique identifier for the account</param>
+        /// <returns>The account</returns>
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(Guid id)
+        {
+            var retrievedAccount = await Orchestrator.GetAccountBy(id);
+
+            return Ok(retrievedAccount);
+        }
+
+        /// <summary>
+        /// Gets all users of an account
+        /// </summary>
+        /// <param name="accountId">The unique identifier for the account</param>
+        /// <returns>The account users</returns>
+        [HttpGet("{accountId:guid}/Users")]
+        public async Task<IActionResult> GetUsers(Guid accountId)
+        {
+            var accounts = await Orchestrator.GetAccountUsers(accountId);
+
+            return Ok(accounts);
+        }
+
+        /// <summary>
+        /// Adds a user to an account
+        /// </summary>
+        /// <param name="accountId">Id of the account</param>
+        /// <param name="userId">Id of the user</param>
+        /// <returns></returns>
+        [HttpPut("{accountId:guid}/User/{userId:guid}")]
+        public async Task RegisterUser(Guid accountId, Guid userId)
+        {
+            await Orchestrator.RegisterUserToAccount(accountId, userId);
+        }
+
+        /// <summary>
+        /// Upgrades an account plan
+        /// </summary>
+        /// <param name="accountId"></param>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPut("{accountId:guid}")]
+        public async Task<IActionResult> UpgradePlan(Guid accountId, [FromBody]UpgradePlanRequest request)
+        {
+            if (!PlanValidator.IsPlanValid(request.Plan))
```

Comment:
It feels that supported plan check should happen inside Orchestrator because this is also part of business logic.

For example, what would this method of AccountOrchestrator return if GetPlanType returns `null` ?

```c#
public async Task<Guid> CreateAccount(CreateAccountRequest request)
        {
            var plan = GetPlanTypeFrom(request);

            var account = new Account(plan);

            return await Repository.Create(account.ToDatabase());
        }
```

Reply:
Makes sense, moved the last bit of it into the domain!